src/core/server/saved_objects/service/lib/errors.ts

```diff
@@ -202,4 +202,23 @@ export class SavedObjectsErrorHelpers {
   public static isGeneralError(error: Error | DecoratedError) {
     return isSavedObjectsClientError(error) && error[code] === CODE_GENERAL_ERROR;
   }
+
+  public static createGenericNotFoundEsUnavailableError(
```

Comment:
Do we want to differentiate this error from `isEsUnavailableError`? I'm inclined to say that we want applications to handle this condition in the same way they would handle any other issue reaching Elasticsearch. I'm not sure if there's anything different a consumer would or can do in this case vs. any other `isEsUnavailableError`. Another benefit of using the existing error is that any existing code that is handling it will also handle this new case.

Curious what @mshustov thinks as well.

Reply:
Agree, it's rather `ES server is unavailable` rather than `a Kibana entity is not found`.
A user reacts to them in different ways: in the case of `a Kibana entity is not found`, the user needs to change adjust the request params and try again.
While in the case of `ES server is unavailable`, the error is not actionable: the user needs to contact an admin to address the availability problem.

Reply:
Initially, I did try throwing `isEsUnavailableError` but that returns a [boolean](https://github.com/elastic/kibana/blob/master/src/core/server/saved_objects/service/lib/errors.ts#L194-L196), not an actual `Error` or `DecoratedError`. If you mean that we should rather throw a [decorated](https://github.com/elastic/kibana/blob/master/src/core/server/saved_objects/service/lib/errors.ts#L190-L192) `EsUnavailableError` that will return `true` on checking that the error is `isEsUnavailable`, then we can do that.
If I'm mis-reading the types here, please let me know!.

Reply:
> f you mean that we should rather throw a decorated EsUnavailableError that will return true on checking that the error is isEsUnavailable, then we can do that.

Yeah, I was thinking about something like this. @joshdover Did you mean the same?

Reply:
`createGenericNotFoundEsUnavailableError` will return a [503](https://github.com/elastic/kibana/pull/107104/files#diff-c6ac0127a77f2ba335589ef96b96203ca6b393d22a1878e95225c912c0e86d51R489) error that adds [the message from a `NotFoundError`](https://github.com/elastic/kibana/pull/107104/files#diff-c6ac0127a77f2ba335589ef96b96203ca6b393d22a1878e95225c912c0e86d51R483) to the `EsUnavailableError`. The decorated error that's thrown will [assert `true` to a check for `isEsUnavailableError`](https://github.com/elastic/kibana/blob/master/src/core/server/saved_objects/service/lib/errors.ts#L195), since the latter returns a `boolean` for the `statusCode` being 503.

Reply:
I initially replied with this but then deleted it again. I'm adding it back for more context on the error that I'm using:

`createGenericNotFoundEsUnavailableError` will return a [503](https://github.com/elastic/kibana/pull/107104/files#diff-c6ac0127a77f2ba335589ef96b96203ca6b393d22a1878e95225c912c0e86d51R489) error that adds [the message from a `NotFoundError`](https://github.com/elastic/kibana/pull/107104/files#diff-c6ac0127a77f2ba335589ef96b96203ca6b393d22a1878e95225c912c0e86d51R483) to the `EsUnavailableError`. The decorated error that's thrown will [assert `true` to a check for `isEsUnavailableError`](https://github.com/elastic/kibana/blob/master/src/core/server/saved_objects/service/lib/errors.ts#L195), since the latter returns a `boolean` for the `statusCode` being 503.